Makefile

```diff
@@ -5,7 +5,17 @@ PATH := $(PWD)/protoc/bin:$(PWD)/bin:$(PATH)
 SHELL := env PATH=$(PATH) /bin/sh
 
 PROTOC_VERSION_NEEDED := 28.2
-PROTOC_VERSION := $(shell command -v protoc > /dev/null 2>&1 && protoc --version)
+PROTOC_VERSION := $(shell command -v protoc > /dev/null 2>&1 && protoc --version || echo 'none')
+PROTOC_BASE_URL := https://github.com/protocolbuffers/protobuf/releases/download/
+
+ARCH := $(shell uname -m)
+ifeq ($(shell uname), Linux)
+	PROTOC_FILENAME := protoc-$(PROTOC_VERSION_NEEDED)-linux-$(ARCH).zip
+endif
+ifeq ($(shell uname), Darwin)
+	PROTOC_FILENAME := protoc-$(PROTOC_VERSION_NEEDED)-osx-$(ARCH).zip
```

Comment:
sadly this generates things like the following:

```
curl --location --output tmp/protoc-28.2-osx-arm64.zip https://github.com/protocolbuffers/protobuf/releases/download//v28.2/protoc-28.2-osx-arm64.zip
```

Though arm one is likely `aarch_64` in the download location, so you'll need to update this. (using `aarch_64` instead worked well with my machine)

Reply:
Thanks for checking--can you try the updated code?

Reply:
worked well 🎉